diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 91a502bb9..2f6d6e2bc 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -148,6 +148,44 @@ else
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
 fi
 cp "$($MAKE -s -f $srctree/Makefile image_name)" "$tmpdir/$installed_image_path"
+cp "${srctree}/debian/cmdline.txt" "$tmpdir/boot"
+cp "${srctree}/debian/config.txt" "$tmpdir/boot"
+cp "${srctree}/debian/boot.cmd" "$tmpdir/boot"
+mkimage -A arm -O linux -T script -C none -n "Boot Script" -d "$tmpdir/boot/boot.cmd" "$tmpdir/boot/boot.scr"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/bootcode.bin -o "$tmpdir/boot/bootcode.bin"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup.dat -o "$tmpdir/boot/fixup.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup4.dat -o "$tmpdir/boot/fixup4.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup4cd.dat -o "$tmpdir/boot/fixup4cd.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup4db.dat -o "$tmpdir/boot/fixup4db.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup4x.dat -o "$tmpdir/boot/fixup4x.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup_cd.dat -o "$tmpdir/boot/fixup_cd.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup_db.dat -o "$tmpdir/boot/fixup_db.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/fixup_x.dat -o "$tmpdir/boot/fixup_x.dat"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start.elf -o "$tmpdir/boot/start.elf"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start4.elf -o "$tmpdir/boot/start4.elf"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start4cd.elf -o "$tmpdir/boot/start4cd.elf"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start4db.elf -o "$tmpdir/boot/start4db.elf"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start4x.elf -o "$tmpdir/boot/start4x.elf"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start_cd.elf -o "$tmpdir/boot/start_cd.elf"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start_db.elf -o "$tmpdir/boot/start_db.elf"
+curl https://raw.githubusercontent.com/raspberrypi/firmware/master/boot/start_x.elf -o "$tmpdir/boot/start_x.elf"
+cp "${srctree}/arch/arm/boot/dts/bcm2710-rpi-2-b.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2710-rpi-3-b.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2710-rpi-cm3.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2710-rpi-zero-2.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2710-rpi-zero-2-w.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2711-rpi-400.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2711-rpi-4-b.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2711-rpi-cm4.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2711-rpi-cm4s.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2708-rpi-zero.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2708-rpi-zero-w.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2708-rpi-b.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2708-rpi-b-plus.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2708-rpi-b-rev1.dtb" "$tmpdir/boot"
+cp "${srctree}/arch/arm/boot/dts/bcm2708-rpi-cm.dtb" "$tmpdir/boot"
+cp -Lr "$srctree/arch/arm64/boot/dts/overlays" "$tmpdir/boot"
 
 if is_enabled CONFIG_OF_EARLY_FLATTREE; then
 	# Only some architectures with OF support have this target
